### PR TITLE
SALTO-2474: fix adapter rename to rename TemplateExpressions as well

### PR DESCRIPTION
**Pull request by noamhammer**
_fix adapter rename to rename TemplateExpressions as well_

---

_Additional context for reviewer_
None

---
_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

**Comment by netama** [comment]
please wait for @yanaise to review this

File: packages/workspace/src/element_adapter_rename.ts [hunk 1]
```diff
@@ -107,6 +115,14 @@ const transformElemIDAdapter = (accountName: string): TransformFunc => async (
       value.refType = fieldType
     }
   }
+}
+
+const transformElemIDAdapter = (accountName: string): TransformFunc => async (
+  { value }
+) => {
+  updateReferenceExpression(value, accountName)
```
Comment (yanaise) [inline packages/workspace/src/element_adapter_rename.ts:123]: nit: To me it makes more sense to have "if isReferenceExpression, if isTemplateExpression, if isElement" in this function, and have updateElement etc only have the code that  changes the object

File: packages/workspace/test/workspace/element_adapter_rename.test.ts [hunk 2]
```diff
@@ -87,6 +92,9 @@ describe('rename adapter in elements', () => {
   })
   const changedInstance = new InstanceElement('InstanceElement', changedObject, {
     field: new ReferenceExpression(changedInnerType.elemID),
+    templateField: new TemplateExpression({
+      parts: ['prefix', new ReferenceExpression(changedInnerType.elemID)],
```
Comment (yanaise) [inline packages/workspace/test/workspace/element_adapter_rename.test.ts:96]: I would have a test (it can be the only test) with more than one ReferenceExpression in the same TemplateExpression, as some edge cases revolve around that.

File: packages/workspace/src/element_adapter_rename.ts [hunk 3]
```diff
@@ -93,6 +92,15 @@ const transformElemIDAdapter = (accountName: string): TransformFunc => async (
...
+}
+
+const updateTemplateExpression = (value: unknown, accountName: string): void => {
+  if (isTemplateExpression(value)) {
+    value.parts.forEach(part => updateReferenceExpression(part, accountName))
+  }
+}
+
+const updateElement = (value: unknown, accountName: string): void => {
   if (isElement(value) && value.elemID.adapter !== accountName) {
```
Comment (yanaise) [inline packages/workspace/src/element_adapter_rename.ts:104]: nit: if you want, you can remove the value.elemID.adapter test because it occurs before the whole transformation (and wcs createAdapterReplacedID just doesn't do anything)

**Review by yanaise — APPROVED** [review]
Approved with 3 minor comments
